### PR TITLE
crimson/os/seastore: fix in check_node

### PR DESCRIPTION
**Pull request by aisakaki**
EXIST_CLEAN and EXIST_MUTATION_PENDING shuold not be treated as CLEAN in check_node because they are transaction private and the leafnode has been duplicated for write.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by cyx1231st** [comment]
I think we can merge this as the direct fix to unblock https://github.com/ceph/ceph/pull/51179, then scan the other places. If it is reasonable to you @xxhdx1985126

File: src/crimson/os/seastore/btree/fixed_kv_btree.h
Comment (xxhdx1985126) [inline src/crimson/os/seastore/btree/fixed_kv_btree.h:525]: should we include `is_exist_clean()` and `is_existing_mutation_pending()` in `is_initial_pending()`?

Comment (cyx1231st) [reply src/crimson/os/seastore/btree/fixed_kv_btree.h:525]: Probably not as `is_initial_pending()`. But I don't see a good name to group them together.

Or use the following decision tree in the lba implementation, putting them in the else section:
* is_stable(): CLEAN_PENDING, CLEAN, DIRTY
* is_pending():
  - is_mutation_pending(): MUTATION_PENDING
  - else: INITIAL_WRITE_PENDING, EXIST_CLEAN, EXIST_MUTATION_PENDING
* else: !is_valid(): INVALID

What do you think?

Comment (xxhdx1985126) [reply src/crimson/os/seastore/btree/fixed_kv_btree.h:525]: > Or use the following decision tree in the lba implementation, putting them in the else section:
> 
> * is_stable(): CLEAN_PENDING, CLEAN, DIRTY
> * is_pending():
>   
>   * is_mutation_pending(): MUTATION_PENDING
>   * else: INITIAL_WRITE_PENDING, EXIST_CLEAN, EXIST_MUTATION_PENDING
> * else: !is_valid(): INVALID

Sorry for the late reply, I think this is fine.

**Comment by cyx1231st** [comment]
Still see issues when remap/overwrite extents, might need to block this until it is complete enough to address the issue.

**Comment by aisakaki** [comment]
confirmed that issue not relate to this~

**Comment by aisakaki** [comment]
jenkins test api

File: src/crimson/os/seastore/btree/fixed_kv_btree.h
Comment (aisakaki) [reply src/crimson/os/seastore/btree/fixed_kv_btree.h]: @xxhdx1985126 @cyx1231st changed the decision path in new push ~

**Comment by aisakaki** [comment]
> Still see issues when remap/overwrite extents, might need to block this until it is complete enough to address the issue.

Confirmed that issues not related to this. Solved in the latest push of PR51179.

**Comment by aisakaki** [comment]
jenkins test make check

File: src/crimson/os/seastore/cached_extent.h
Comment (cyx1231st) [inline src/crimson/os/seastore/cached_extent.h:418]: /// Returns true if extent is stable and shared among transactions

**Review by cyx1231st — APPROVED** [review]

File: src/crimson/os/seastore/cached_extent.h
Comment (aisakaki) [reply src/crimson/os/seastore/cached_extent.h]: done

**Pull request merged**
